OpenRA.Mods.Common/Pathfinder/PathCacheStorage.cs

```diff
@@ -9,12 +9,67 @@
  */
 #endregion
 
+using System;
 using System.Collections.Generic;
 using System.Linq;
 
 namespace OpenRA.Mods.Common.Pathfinder
 {
-	public class PathCacheStorage : ICacheStorage<List<CPos>>
+	public enum PathQueryType : byte
+	{
+		UnitPath,
+		UnitPathToRange
+	}
+
+	public struct PathCacheKey
```

Comment:
this should be readonly othewise a copy will be made when using the in param, see https://docs.microsoft.com/en-us/dotnet/csharp/write-safe-efficient-code